src/handlers/flow/FlowStreamHandler.h

```diff
@@ -78,87 +82,105 @@ struct FlowPacket {
-        , topOutIfIndex(FLOW_SCHEMA, "interface", {"top_out_interfaces_" + metric}, "Top output interfaces by " + metric)
         , topGeoLoc(FLOW_SCHEMA, "geo_loc", {"top_geoLoc_" + metric}, "Top GeoIP locations by " + metric)
         , topASN(FLOW_SCHEMA, "asn", {"top_ASN_" + metric}, "Top ASNs by IP by " + metric)
     {
     }
 
     void set_settings(size_t topn_count, uint64_t percentile_threshold)
     {
-        topSrcIP.set_settings(topn_count, percentile_threshold);
-        topDstIP.set_settings(topn_count, percentile_threshold);
-        topSrcPort.set_settings(topn_count, percentile_threshold);
-        topDstPort.set_settings(topn_count, percentile_threshold);
-        topSrcIPandPort.set_settings(topn_count, percentile_threshold);
-        topDstIPandPort.set_settings(topn_count, percentile_threshold);
+        topInSrcIP.set_settings(topn_count, percentile_threshold);
+        topInDstIP.set_settings(topn_count, percentile_threshold);
+        topOutSrcIP.set_settings(topn_count, percentile_threshold);
+        topOutDstIP.set_settings(topn_count, percentile_threshold);
+        topInSrcPort.set_settings(topn_count, percentile_threshold);
+        topInDstPort.set_settings(topn_count, percentile_threshold);
+        topOutSrcPort.set_settings(topn_count, percentile_threshold);
+        topOutDstPort.set_settings(topn_count, percentile_threshold);
+        topInSrcIPandPort.set_settings(topn_count, percentile_threshold);
+        topInDstIPandPort.set_settings(topn_count, percentile_threshold);
+        topOutSrcIPandPort.set_settings(topn_count, percentile_threshold);
+        topOutDstIPandPort.set_settings(topn_count, percentile_threshold);
         topConversations.set_settings(topn_count, percentile_threshold);
-        topInIfIndex.set_settings(topn_count, percentile_threshold);
-        topOutIfIndex.set_settings(topn_count, percentile_threshold);
         topGeoLoc.set_settings(topn_count, percentile_threshold);
         topASN.set_settings(topn_count, percentile_threshold);
     }
 };
 
-struct FlowDevice {
+struct Counters {
+    Counter UDP;
+    Counter TCP;
+    Counter OtherL4;
+    Counter IPv4;
+    Counter IPv6;
+    Counter total;
+    Counters(std::string direction, std::string metric)
+        : UDP(FLOW_SCHEMA, {direction + "_udp_" + metric}, "Count of " + direction + " UDP by " + metric)
+        , TCP(FLOW_SCHEMA, {direction + "_tcp_" + metric}, "Count of " + direction + " TCP by " + metric)
+        , OtherL4(FLOW_SCHEMA, {direction + "_other_l4_" + metric}, "Count of " + direction + " " + metric + " which are not UDP or TCP")
+        , IPv4(FLOW_SCHEMA, {direction + "_ipv4_" + metric}, "Count of " + direction + " IPv4 by " + metric)
+        , IPv6(FLOW_SCHEMA, {direction + "_ipv6_" + metric}, "Count of " + direction + " IPv6 by " + metric)
+        , total(FLOW_SCHEMA, {direction + "_" + metric}, "Count of total " + direction + " flows by " + metric)
```

Comment:
Does this replace the previous `flow_records_total`? metric? Not sure we need the direction here, but if it comes for free... we can work with it